stan/math/rev/arr/functor/coupled_ode_system.hpp

```diff
@@ -45,7 +45,7 @@ struct coupled_ode_system<F, double, var> {
   const F& f_;
   const std::vector<double>& y0_dbl_;
   const std::vector<var>& theta_;
-  const std::vector<double> theta_dbl_;
+  std::vector<var> theta_copy_;
```

Comment:
Could we please rename this to something descriptive? This is meant to be a copy of the `theta`, but placed on the global stack, but essentially treated as temporaries where we can reset the adjoints. I don't know what a good name is, but just calling it `theta_copy_` really isn't that great.

The old name was ok because it implies we've lost the connections between `theta_` and `theta_dbl_` by demoting it to a `double`.